doc/manual/parallel-computing.rst

```diff
@@ -455,21 +455,131 @@ points: in this case, when :func:`remotecall_fetch` is called.
-until it is emptied.
+The section on Tasks in :ref:`man-control-flow`, discussed the execution of
+multiple functions in a co-operative manner. :class:`Channels` can be quite useful
+to pass data between running tasks, particularly those involving I/O operations.
+
+Examples of operations involving I/O include reading/writing to files,
+accessing web services, executing external programs, etc. In all
+these cases, overall execution time can be improved if other tasks can be run
+while a file is being read, or while waiting for an external service/program to complete.
+
+A channel can be visualized as a pipe, i.e., it has a write end and read end.
+- Multiple writers in different tasks can write to the same channel concurrently via
+  :func:`put!` calls.
+- Multiple readers in different tasks can read data concurrently via :func:`take!` calls.
+- As an example::
+
+    # Given Channels c1 and c2,
+    c1 = Channel(32)
+    c2 = Channel(32)
+
+    # and a function `foo()` which reads items from from c1, processes the item read
+    # and writes a result to c2,
+    function foo()
+        while true
+            data = take!(c1)
+            .......             # process data
+            put!(c2, result)    # write out result
+        end
+    end
+
+    # we can schedule `n` instances of `foo()` to be active concurrently.
+    for _ in 1:n
+        @schedule foo()
+    end
+
+- Channels are created via the :class:`Channel{T}(sz)` constructor. The channel will only hold
+  objects of type ``T``. If the type is not specified, the channel can hold objects of
+  any type. ``sz`` refers to the maximum number of elements that can be held in the channel
+  at any time. For example, ``Channel(32)`` creates a channel that can hold a maximum of 32 objects
+  of any type. A ``Channel{MyType}(64)`` can hold upto 64 objects of ``MyType`` at any time.
+- If a :class:`Channel` is empty, readers (on a :func:`take!` call) will block till data is available.
+- If a :class:`Channel` is full, writers (on a :func:`put!` call) will block till space becomes available.
+- :func:`isready` tests for the presence of any object in the channel, while :func:`wait`
+  waits for an object to become available.
+- A :class:`Channel` is in an open state initially. This means that it can be
+  read and written to freely via :func:`take!` and :func:`put!` calls. :func:`close` closes a :class:`Channel`.
+  On a closed :class:`Channel`\, :func:`put!` will fail. For example::
+
+        julia> c=Channel(2);
+
+        julia> put!(c,1);   # `put!` on an open channel succeeds
+
+        julia> close(c);
+
+        julia> put!(c,2);   # `put!` on a closed channel throws an exception.
+        ERROR: InvalidStateException("Channel is closed.",:closed)
+        in put!(::Channel{Any}, ::Int64) at ./channels.jl:77
+
+
+- :func:`take!` and :func:`fetch` on a closed channel successfully return any existing values till
```

Comment:
sure but this is IIRC the first place `fetch` is mentioned in this file, and there's no context about it